### ISSUE TITLE
Support for custom encoders

### ISSUE THREAD

**Issue by zbjornson** [darrachequesne/notepack#18]
msgpack-lite supports [Custom Extension Types (Codecs)](https://github.com/kawanet/msgpack-lite#custom-extension-types-codecs). There's a project that I'd love to switch to this faster lib, but it uses a codec.

(Thanks for maintaining this fork btw!)

**Comment by Domiii** [comment]
Agreed. Not being able to encode basic built-ins, such as `Symbol`, is a big downer. Probably will have to fork our own versions to patch that in...
